src/main/java/edu/harvard/iq/dataverse/authorization/AuthenticationServiceBean.java

```diff
@@ -511,6 +533,69 @@ public AuthenticatedUser lookupUser( String apiToken ) {
         return tkn.getAuthenticatedUser();
     }
     
+    /*
+    getDeleteUserErrorMessages( AuthenticatedUser au )
+    method which checks for reasons that a user may not be deleted
+    -has created dvObjects
+    -has roles
+    -has guestbook records
+    -etc.
+    An empty string is returned if the user is 'deletable'
+    */
+    
+    public String getDeleteUserErrorMessages(AuthenticatedUser au) {
+        String retVal = "";
+        if (!dvObjSvc.findByAuthenticatedUserId(au).isEmpty()) {
+            retVal += BundleUtil.getStringFromBundle("admin.api.deleteUser.failure.dvobjects");
+        }
+
+        if (!roleAssigneeSvc.getAssignmentsFor(au.getIdentifier()).isEmpty()) {
+            if (!retVal.isEmpty()) {
+                retVal += "; ";
+            }
+            retVal += BundleUtil.getStringFromBundle("admin.api.deleteUser.failure.roleAssignments");
+        }
+
+        if (!gbRespSvc.findByAuthenticatedUserId(au).isEmpty()) {
+            if (!retVal.isEmpty()) {
+                retVal += "; ";
+            }
+            retVal += BundleUtil.getStringFromBundle("admin.api.deleteUser.failure.gbResps");
+        }
+
+        if (!datasetVersionService.getDatasetVersionUsersByAuthenticatedUser(au).isEmpty()) {
+            if (!retVal.isEmpty()) {
+                retVal += "; ";
+            }
+            retVal += BundleUtil.getStringFromBundle("admin.api.deleteUser.failure.versionUser");
+        }
+
+        if (!explicitGroupService.findGroups(au).isEmpty()) {
+            if (!retVal.isEmpty()) {
+                retVal += "; ";
+            }
+            retVal += BundleUtil.getStringFromBundle("admin.api.deleteUser.failure.groupMember");
+        }
+
+        if (userHasPendingAccessRequests(au)) {
```

Comment:
Currently, once an access request is acted on, we delete, so I think it would be fair to just also delete these requests when you delete the user (instead of failing). My one question is that there has been talk of storing these for historical purposes. If/when we do, we may want to then either a) not allow delete of user or b) anonymize the requests. Or c) still delete them, of course, I just think we'll need to think of that case.

But that said, it makes sense to change that delete to be cascading now.

Reply:
Deleting requests, explicit group memberships and world map tokens in a newly created method that is run if there are no other records that would prevent a deletion (created dataset, downloaded a file etc.)